### PR TITLE
Add unit tests for mapper and problems attempted

### PR DESCRIPTION
**Pull request by mulby**
@brianhw @jab5569

### CODE REVIEW

File: edx/analytics/tasks/tests/test_student_engagement.py [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+        self.task.init_local()
+
+    def _check_output(self, inputs, column_num, expected_value):
+        """Compare generated with expected output."""
+        output = self._get_reducer_output(inputs)
+        self.assertEquals(len(output), 1)
+        self.assertEquals(output[0][column_num], expected_value)
+
+    def test_no_events(self):
+        self._check_output([], self.WAS_ACTIVE_COLUMN_NUM, 1)
```
Comment (brianhw) [inline edx/analytics/tasks/tests/test_student_engagement.py]: Why do we expect the output to have a count of 1 when the input contained no events for the given key?

**Comment by brianhw** [comment]
Just one small question. LGTM otherwise.

**Comment by mulby** [comment]
@brianhw I put in a fix for your comment. Ready for re-review.

**Comment by brianhw** [comment]
Missing a test for last-subsection-viewed with no timestamp.

**Comment by mulby** [comment]
I actually couldn't figure out how to reach that code, I suspect this "if" statement can never actually get executed (given that invalid timestamps are filterd out by the event log selection process).

We should probably remove it, however, I like defensive stuff like this, since those other parts of the code might change which would allow invalid timestamps to seep into this area...

**Comment by brianhw** [comment]
I'm fine with that.  :+1:

**Pull request merged**